NOnion/Crypto/HiddenServicesCipher.fs

```diff
@@ -42,7 +42,7 @@ module HiddenServicesCipher =
         let output = Array.zeroCreate length
 
         digestEngine.BlockUpdate(data, 0, data.Length)
-        digestEngine.DoFinal(output, 0) |> ignore<int>
+        digestEngine.DoFinal(output, 0, length) |> ignore<int>
```

Comment:
btw should we change param to uint?

Reply:
Every call to this function uses constant numbers, changing it to uint just adds more "casts" without any actual safety benefit (because all of the numbers are constants).

Reply:
> why not make those constant numbers uint?

Because then I have to go into a rabbit hole of changing everywhere those constants are used to use uint also.

Reply:
let's do that in a final refactoring commit :)